sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalResourceId.java

```diff
@@ -35,7 +35,8 @@
  */
 class LocalResourceId implements ResourceId {
 
-  private final Path path;
+  private final String pathString;
+  private transient volatile Path cachedPath;
```

Comment:
Since `Path` is not serializable, it has to be `transient`. Instead, we store `pathString` and use `Paths.get(pathString)` whenever we need the `Path`. (Like `TableSchema` in `BigQueryIO`). The cached copy is just to avoid frequent calls to `Paths.get()` in the event that function is slow.